src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -405,20 +405,23 @@ ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
                                        uct_ib_mlx5_mmio_mode_t mmio_mode)
 {
 #if HAVE_DEVX
-    uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
-                                                UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
+    uar->uar = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
+                                     UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
     if (uar->uar == NULL) {
-        if (UCT_IB_MLX5_UAR_ALLOC_TYPE_WC == UCT_IB_MLX5_UAR_ALLOC_TYPE_NC) {
-            ucs_error("mlx5dv_devx_alloc_uar(WC) failed: %m");
-            return UCS_ERR_NO_MEMORY;
-        }
-
-        uar->uar        = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
-                                                UCT_IB_MLX5_UAR_ALLOC_TYPE_NC);
+#if HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC
+        ucs_debug("UAR alloc WC failed. Fallback to NC\n");
```

Comment:
ucs_debug("mlx5dv_devx_alloc_uar(device=%s, flags=0x%x) failed: %m, fallback to NC\n");